### PR TITLE
Makefile overhaul, Refactor test scripts

### PR DESCRIPTION
**Pull request by barney-s**
Redo makefile
Fix lint issues
Fix misspell
Auto fix license
Refactor test scripts

### CODE REVIEW

**Comment by barney-s** [comment]
@nan-yu @mortent - Please review

File: Makefile [hunk 1]
```diff
@@ -34,106 +26,177 @@ WEBHOOK_ROOT ?= $(MANIFEST_ROOT)/webhook
...
+license: $(TOOLBIN)/addlicense
+	$(TOOLBIN)/addlicense  -y $(shell date +"%Y") -c "The Kubernetes Authors." -f LICENSE_TEMPLATE .
+
+.PHONY: tidy
+tidy:
+	go mod tidy
+
+.PHONY: lint
+lint: $(TOOLBIN)/golangci-lint
+	$(TOOLBIN)/golangci-lint run ./api/... ./controllers/...
```
Comment (nan-yu) [inline Makefile]: I would prefer running glangci-lint on all packages `./...`, as `main.go` is missing from `./api/... ./controllers/...`

Comment (barney-s) [reply Makefile]: ack will do.

**Comment by barney-s** [comment]
@mortent 
@janetkuo 
please review

**Pull request merged**